src/js/appServices.ts

```diff
@@ -262,15 +269,45 @@ export class AppServices implements IAppServices {
     }
 
     getApiHeaders(apiUrl:string):HTTPHeaders {
-        const prefixes = Object.keys(this.apiHeadersMapping);
-        for (let i = 0; i < prefixes.length; i += 1) {
-            if (apiUrl && apiUrl.indexOf(prefixes[i]) === 0) {
-                return this.apiHeadersMapping[prefixes[i]];
+        const srchHeaders = (location:{[url:string]:HTTPHeaders}) => {
+            const srch = pipe(
+                location,
+                Dict.toEntries(),
+                List.find(([url,]) => apiUrl.indexOf(url) === 0)
+            );
+            if (srch !== undefined) {
+                return srch[1];
             }
+            return undefined;
+        };
+        const srch1 = srchHeaders(this.getDynamicApiKeyHeaders());
```

Comment:
maybe we should merge those two srch1, srch2 headers so we can use both some static headers and dynamic ones at the same time